doc/man1/openssl-pkeyutl.pod.in

```diff
@@ -133,22 +140,32 @@ The input is a certificate containing a public key.
 =item B<-rev>
 
 Reverse the order of the input buffer. This is useful for some libraries
-(such as CryptoAPI) which represent the buffer in little endian format.
+(such as CryptoAPI) which represent the buffer in little-endian format.
 This cannot be used in conjunction with B<-rawin>.
 
 =item B<-sign>
 
-Sign the input data (which must be a hash) and output the signed result. This
-requires a private key.
+Sign the input data and output the signed result. This requires a private key.
+Using a message digest operation along with this is recommended -
+see the B<-rawin> and B<-digest> options for details.
+Otherwise, the input data given with the B<-in> option is assumed to already
+be a digest, but this may then require an additional B<-pkeyopt> C<digest:>I<md>
+in some cases (e.g., RSA with the default PKCS#1 padding mode).
+Even for other algorithms like ECDSA, where the additional B<-pkeyopt> option
+does not affect signature output, it recommended to use
+to sanity-check that the input length is consistent with the purported digest.
```

Comment:
```suggestion
does not affect signature output, it is recommended, as it makes it possible
to check that the input length is consistent with the intended digest.
```

Reply:
Done, simplifying to:
> does not affect signature output, it is recommended, as it enables
checking that the input length is consistent with the intended digest.